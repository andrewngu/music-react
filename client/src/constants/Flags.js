// import rox-browser from npm
import Rox from 'rox-browser' 
import UserRepo from  '../store/UserRepo'

// define an exported object that contains the flags
const Flags = {
  followingView: new Rox.Flag(),
  history: new Rox.Flag(),
  shuffle: new Rox.Flag(),
  repeat: new Rox.Flag(),
  startFollowingWord: new Rox.Variant('Follow', ['Follow', 'Start Following', 'Watch Him'])
};

//  Register the flags object under the namespace "default"
Rox.register('default', Flags);
const fakeUserID =  "" + Math.random();
Rox.setCustomStringProperty('rox.distinct_id', fakeUserID);
analytics.identify(fakeUserID, {});
// setup the SDK with the production environment key of app soundredux - see https://app.rollout.io/app/5b5ea80aec73e3454653ecc4
Rox.setup('5bb493fcbae7137e6c9dea86', {
  impressionHandler: (reporting, experiment) => {
    if (experiment && experiment.labels.includes('segment')){
	let props = {};
        props[`rollout_${reporting.name}`] = reporting.value;
        props[`rollout_${reporting.name}_labels`] = experiment.labels.join(',');
        analytics.identify(fakeUserID, props);
    }
  }
});

// For demo purposes we will etch the configuration from the server every 4 seoconds 
setInterval( function(){
  Rox.fetch();
}, 4000);

// Dummy user state
Rox.setCustomStringProperty('plan', () => UserRepo.getUser().plan);
Rox.setCustomStringProperty('email', () => UserRepo.getUser().email);
Rox.setCustomNumberProperty('playlist_count', () => UserRepo.getUser().playlistCount);
Rox.setCustomStringProperty('soundcloud_id', () => UserRepo.getUser().id);

export default Flags;
