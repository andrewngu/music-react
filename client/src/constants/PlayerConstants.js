export const PLAYER_PLAY_PREVIOUS_SONG_HINT = 'Play previous';
export const PLAYER_PLAY_CURRENT_SONG_HINT = 'Play current';
export const PLAYER_PLAY_NEXT_SONG_HINT = 'Play next';
export const PLAYER_REPEAT_PLAY_HINT = 'Repeat';
export const PLAYER_SHUFFLE_HINT = 'Shuffle';
export const PLAYER_RECENTLY_PLAYED_HINT = 'Recently played';
