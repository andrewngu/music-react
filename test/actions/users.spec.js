jest.mock('global', () => ({
  fetch: jest.fn(url => Promise.resolve({
    json: () => {
      const tracks = [];
      const followings = {collection: []};
      const profiles =  [];

      const calls = {
        tracks: new RegExp('/users/100/tracks'),
        followings: new RegExp('/users/100/followings'),
        profiles: new RegExp('/users/100/web-profiles'),
      };
      const responses = { tracks, followings, profiles };

      if (url.match(calls.tracks)) {
        return Promise.resolve(responses.tracks);
      }
      if (url.match(calls.followings)) {
        return Promise.resolve(responses.followings);
      }
      if (url.match(calls.profiles)) {
        return Promise.resolve(responses.profiles);
      }
      return Promise.reject(new Error('unexpected fetch'));
    },
  })),
}));

import { fetch } from 'global';
import * as actions from '../../scripts/actions/UsersActions';
import * as types from '../../scripts/constants/ActionTypes';
import { mockStore } from '../TestUtils';

describe('users actions', () => {
    describe('fetchUserIfNeeded', () => {
        it('should not fetch if user is already loaded', async () => {
            const store = mockStore({entities: {users: {100: {id: 100, description: 'foo', followings: [2, 3, 4]}}}}, []);
            await store.dispatch(actions.fetchUserIfNeeded(100));
            expect(fetch).not.toHaveBeenCalled();
        });

        it('should not fetch if user is already loaded', async () => {
            const store = mockStore(
              { entities: { users: { 100: { id: 100, description: 'foo' }}}},
              [
                { entities: {}, futureUrl: undefined, nextUrl: null, playlist: 'undefined|user', songs: [], type: 'RECEIVE_SONGS' },
                { entities: { users: { 100: { followings: [] } } }, type: 'RECEIVE_USER_FOLLOWINGS' },
                { entities: { users: { 100: { profiles: [] } } }, type: 'RECEIVE_USER_PROFILES' }
              ]
            );
            await store.dispatch(actions.fetchUserIfNeeded(100));
            expect(fetch.mock.calls.map(i => i.join(','))).toMatchSnapshot();
        });
    });

    describe('receiveUserFollowings', () => {
        it('should create RECEIVE_USER_FOLLOWINGS action', () => {
            const entities = {
                users: {
                    1: {
                        id: 1,
                        followings: [2, 3, 4]
                    },
                    2: {id: 2},
                    3: {id: 3},
                    4: {id: 4}
                }
            };
            expect(actions.receiveUserFollowings(entities)).toEqual({
                type: types.RECEIVE_USER_FOLLOWINGS,
                entities
            });
        });
    });

    describe('receiveUser', () => {
        it('should create RECEIVE_USER action', () => {
            const entities = {users: { 200: {id: 200, username: 'kygo'}}};
            expect(actions.receiveUser(entities)).toEqual({type: types.RECEIVE_USER, entities});
        });
    });

    describe('receiveUserProfiles', () => {
        it('should create RECEIVE_USER_PROFILES action', () => {
            const entities = {users: { 1: { profiles: [{service: 'facebook'}, {service: 'twitter'}]}}};
            expect(actions.receiveUserProfiles(entities)).toEqual({type: types.RECEIVE_USER_PROFILES, entities});
        });
    });

    describe('requestUser', () => {
        it('should create REQUEST_USER action', () => {
            const userId = 200;
            expect(actions.requestUser(userId)).toEqual({type: types.REQUEST_USER, userId});
        });
    });
});
