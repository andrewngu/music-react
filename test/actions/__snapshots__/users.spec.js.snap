// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users actions fetchUserIfNeeded should not fetch if user is already loaded 1`] = `
Array [
  "//api.soundcloud.com/users/100/tracks?client_id=f4323c6f7c0cd73d2d786a2b1cdae80c",
  "//api.soundcloud.com/users/100/followings?client_id=f4323c6f7c0cd73d2d786a2b1cdae80c",
  "//api.soundcloud.com/users/100/web-profiles?client_id=f4323c6f7c0cd73d2d786a2b1cdae80c",
]
`;
